src/middleware/current_user.rs

```diff
@@ -1,104 +1,37 @@
+//! Middleware that captures the `user_id` from the signed session cookie
+//!
+//! Due to lifetimes it is not possible to call `mut_extensions()` while a reference obtained from
+//! `extensions()` is still live.  The call to `conduit_cookie::RequestSession::session` needs
+//! mutable access to the request and its extensions, and there is no read-only alternative in
+//! `conduit_cookie` to access the session cookie.  This means that it is not possible to access
```

Comment:
> Would our code be better if there was a read-only alternative to get the session cookie?

It would be nice to support that upstream in `conduit-cookie`.  We only use the session cookie here and during the login oauth workflow, and in both cases we don't need a database connection at that particular moment so it hasn't been a problem, yet.  Although I don't really see us adding more complex cookie logic anytime soon.

> It looks like the write access is used for caching the results of decoding the cookie.

As I understand it, the middleware layer from `conduit-cookie` decodes the cookie before the request is processed by the endpoint, and then the middleware adds the (possibly updated) cookie back into every response.  So the `&mut HashMap<_, _>` returned here is what provides the API for mutating the session cookie.

There are some changes (especially with respect to bumping old deps, and more especially wrt our very old dependency on `semver`) that I would like make to the `conduit-*` ecosystem.  Last I looked we (crates.io) are the only public users of these crates, so it would be awesome if we could make some improvements and release new versions of these crates.  Longer term, it would be nice if we could add async/await support in our middleware stack.

For now, I think its fine to document the workaround.  This also provides an easy way for the test harness to simulate cookie authenticated requests, although it would certainly be nice to construct real cookies in the test harness instead.